pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -96,7 +100,32 @@ type RegistryPoll struct {
 	// Interval is used to determine the time interval between checks of the latest catalog source version.
 	// The catalog operator polls to see if a new version of the catalog source is available.
 	// If available, the latest image is pulled and gRPC traffic is directed to the latest catalog source.
-	Interval *metav1.Duration `json:"interval,omitempty"`
+	RawInterval  string           `json:"interval,omitempty"`
+	Interval     *metav1.Duration `json:"-"`
+	ParsingError string           `json:"-"`
```

Comment:
Why is the Parsing error written to the spec? I assume its because it isn't possible to bubble this up and write it to the status?

I understand that following suggestion may not be viable for a bug fix, but I strongly believe that this would be better addressed through use of an webhook that sets sane defaults or rejects invalid requests, as described [here](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html).

Reply:
If a CatalogSource already exists with an invalid interval, what happens to it when the new version of OLM, containing said webhook, is installed?

The `json:"-"` tag indicates the field shouldn't be (un)marshaled, so they're not actually visible to clients of the k8s API. The end effect is that the end user/client both sees no new fields and loses no old fields; on the flip-side, OLM gets a new field to persist extra internal info about field parsing issues in memory.

This is very backportable.